models/dashboards/pevr/pbi_tables/pevr_indicateur_des.sql

```diff
@@ -16,49 +16,66 @@ You should have received a copy of the GNU Affero General Public License
 along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #}
 {{ config(alias="indicateur_des") }}
-
-
 with
-    src as (
+    -- Jumelage du perimetre élèves avec la table mentions
+    perimetre as (
         select
             sch.annee,
             sch.annee_scolaire,
             src.fiche,
             sch.school_friendly_name,
-            case
-                when mentions.ind_reus_sanct_charl = 'O' then 1.0 else 0.0
-            end as 'ind_obtention'
+            mentions.ind_obtention,
+            row_number() over (
+                partition by src.fiche, sch.annee order by sch.annee desc
+            ) as seqid
         from {{ ref("stg_perimetre_eleve_diplomation_des") }} as src
-        left join {{ ref("i_e_ri_mentions") }} as mentions on src.fiche = mentions.fiche
         inner join {{ ref("dim_mapper_schools") }} as sch on src.id_eco = sch.id_eco
+        left join
+            {{ ref("fact_ri_mentions") }} as mentions
+            on src.fiche = mentions.fiche
+            and sch.annee = mentions.annee
         where
-            mentions.prog_charl = '6200'
-            and sch.annee
+            sch.annee
             between {{ store.get_current_year() }}
             - 3 and {{ store.get_current_year() }}
+            and mentions.indice_des = 1.0  -- dip DES
     ),
 
+    -- Ajout des filtres utilisés dans le tableau de bord.
     _filtre as (
         select
-            src.annee,
-            src.annee_scolaire,
-            src.fiche,
-            src.school_friendly_name,
-            src.ind_obtention,
-            ele.genre,
-            y_stud.plan_interv_ehdaa,
-            y_stud.population,
+            perim.annee,
+            perim.annee_scolaire,
+            perim.fiche,
+            case
+                when perim.school_friendly_name is null
+                then '-'
+                else perim.school_friendly_name
+            end as school_friendly_name,
+            perim.ind_obtention,
+            case when ele.genre is null then '-' else ele.genre end as genre,
+            case
+                when y_stud.plan_interv_ehdaa is null
+                then '-'
+                else y_stud.plan_interv_ehdaa
+            end as plan_interv_ehdaa,
+            case
```

Comment:
Comment peux tu avoir une population null sachant que :
1- la fact_yearly_student repose sur la spine
2- tu fais un  inner join fact_yearly_student

Reply:
en fait, c'est plus un filet de sauvetage pour les colonnes qui vont être dans l'étape suivante dans le cube du group by parce le group by cube crée des nulles (correspond au Tout) et pour ne pas risquer des les confondre avec des nulles qui provienne de la saisie de données. mais c'est vrai que la population est un champ que nous avons construit et ne contient pas de nulles

Reply:
je pense ca a mal flaggé les lignes ou je m'interroge. 

L62 à 64 : 
case
when y_stud.population is null then '-' else y_stud.population
end as population,

C'est ici que je le comprends pas. pour le reste c'est ok

Reply:
Mohammed l'a bien dit. J'ai parlé de la population dans un autre commentaire, mais c'est exactement ça.

Reply:
On veut éviter qu'un nulle passe dans le surogate Key. Si ça arrive, on perd la donnée vu que la key généré contient un nulle -> n'est pas dans la table de filtre

Reply:
on fait tu va le perdre bien avant le surogate Key il va etre perdu dans le  coalesce(agg_dip.population, 'Tout') as population